File: Shopping-List-project/shopping_list.py
# shopping_list.py

def main():
    print("Enter up to 5 items to buy:")

    shopping_list = []

    for i in range(5):
        item = input(f"Item {i + 1}: ").strip()
        if item:
            shopping_list.append(item)
        else:
            break  # stop input if user presses enter without typing

    print("\nYour shopping list:")
    print(", ".join(shopping_list))

    print(f"\nYou entered {len(shopping_list)} item(s).")

    # Extra: Search for an item
    search_item = input("\nSearch for an item: ").strip()
    if search_item in shopping_list:
        print(f"{search_item} is in your shopping list!")
    else:
        print(f"{search_item} is NOT in your shopping list.")

if __name__ == "__main__":
    main()

File: Shopping-List-project/shopping-list.js
// shopping_list.js

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const shoppingList = [];
let itemCount = 0;

console.log("Enter up to 5 items to buy:");

function askItem() {
  if (itemCount >= 5) {
    displayList();
    return;
  }

  rl.question(`Item ${itemCount + 1}: `, (input) => {
    const item = input.trim();

    if (item !== '') {
      shoppingList.push(item);
      itemCount++;
      askItem();
    } else {
      displayList(); // Stop early if input is blank
    }
  });
}

function displayList() {
  console.log("\nYour shopping list:");
  console.log(shoppingList.join(", "));

  console.log(`\nYou entered ${shoppingList.length} item(s).`);

  // Extra: search feature
  rl.question("\nSearch for an item: ", (searchTerm) => {
    if (shoppingList.includes(searchTerm.trim())) {
      console.log(`${searchTerm} is in your shopping list!`);
    } else {
      console.log(`${searchTerm} is NOT in your shopping list.`);
    }
    rl.close();
  });
}

askItem();
